apps/public/public.go

```diff
@@ -531,6 +532,33 @@ func WithLoginHint(username string) interface {
 	}
 }
 
+// WithDomainHint pre-populates the login prompt with the IdP domain.
+func WithDomainHint(domain string) interface {
```

Comment:
Nit: Is it possible to refactor our code base so that we wouldn't need to duplicate this code snippet twice?

CC: @chlowell

Reply:
I refactored to avoid the code duplication

Reply:
Oh, thanks for consolidating the new WithDomainHint() and existing WithLoginHint() with a common underlying helper. My original comment was about the WET between confidential client and public client, though. All the implementation and their corresponding test case are currently duplicated in the two clients, while the behavior could ideally be shared among a common base of both confidential client and public client.

Not sure how much refactoring would be needed. I'll defer this topic to @chlowell .

Reply:
I think `withHint()` and its supporting type make the code more complex for no benefit--this approach doesn't lower the maintenance burden. It's better for `WithLoginHint()` and `WithDomainHint()` to simply set their respective fields directly.

Sharing one implementation of `WithDomainHint()` between confidential and public clients might be possible, but the constraints imposed by Go and the current implementation make it difficult and much more complex than parallel options in each package. I don't think it's worth pursuing.

Regarding a shared base for both client types, we do have one, but testing it isn't a substitute for testing the public API.

Reply:
Ok, I reverted the `WithHint()` code and now have separate `WithLoginHint()` and `WithDomainHint()`